### DISCUSSION TITLE
Wrong Data in some Stock Files

### DISCUSSION THREAD

**Discussion by cotton1234** [BennyThadikaran/eod2#210] [General]
Hello Benny,

Lets say i come across a stock which data is wrong.
How to correct it?

Example adroitinfo.csv
Data is coming from 2015 where as the data in broker terminal is from 2017.

Thanks
Saurabh

**Comment by BennyThadikaran** [comment]
Hi Saurabh,

There some rare instances where the stock was permanently suspended and the ISIN was reassigned to another stock.

I had partially resolved it by looking for stocks with missing data for more than 365. Either it escaped the filtering process or there may be a reason i skipped it.

I will have a look at this particular instance tomorrow and let you know.

You can simply remove the excess data if you're sure.
 
Regards,
Benny Thadikaran

**Reply by cotton1234** [reply]
Hello Benny, 

I have developed a script to fetch the data from broker terminal and create a file same as in eod2. 
Will check today i get any error for the same.

If its fine, will use to correct the data as needed for other stocks.

But would it be issue if i update the same with your repo of eod2 and need to do the same for the corrected stock file.?

Thanks
Saurabh

**Comment by cotton1234** [comment]
Got the following Error.

(.venv) saurabhgarg@MacBook-Pro src % python3 init.py
sh: color: command not found
INFO: 12-12-2024 18:35 - __main__ - Downloading Files
INFO: 12-12-2024 18:35 - defs.defs - Starting Data Sync
ERROR: 12-12-2024 18:35 - __main__ - Error during data sync.
Traceback (most recent call last):
  File "/Users/saurabhgarg/shivshakti/eod2/src/init.py", line 114, in <module>
    defs.updateNseEOD(BHAV_FILE, DELIVERY_FILE)
  File "/Users/saurabhgarg/shivshakti/eod2/src/defs/defs.py", line 535, in updateNseEOD
    dq = t.TtlTradgVol if t.SctySrs in ("BE", "BZ") else int(dq)
                                                         ^^^^^^^
ValueError: cannot convert float NaN to integer
INFO: 12-12-2024 18:35 - defs.defs - Rolling back changes from 2024-12-12: /Users/saurabhgarg/shivshakti/eod2/src/eod2_data/daily
INFO: 12-12-2024 18:35 - defs.defs - Rollback successful

Restored the file and it completed fine.

Will check my script.

Thanks
Saurabh

**Comment by BennyThadikaran** [comment]
I am checking adroitinfo.csv in the daily folder and mine starts from `2015-10-13`. On Zerodha Kite, the data starts exactly the same. (Zerodha has some data going back to 2007 though this is not relevant for us)

![Screenshot_2024-12-12_23-00-44](https://github.com/user-attachments/assets/a2320acf-64b7-4dda-b32d-a610fd8be9ce)

If you're seeing additional data, its likely your eod_data folder is not updated. 

See the install instructions to get the latest eod_data. https://github.com/BennyThadikaran/eod2/wiki/Installation#update-eod2_data-folder

If i recall there were only 1 or 2 stocks like this. Let me know if you come across any.

**Reply by cotton1234** [reply]
![image](https://github.com/user-attachments/assets/8d8e1640-2bfe-41a6-97a6-3504cfe00c9d)

I use Fyers, its coming from 2017. Also If you see the data in csv file vs the chart, OHLC data of 2015 is wrong. 

I was able to modify the my script code and update the stock in local repo. 

Thanks
Saurabh

**Reply by BennyThadikaran** [reply]
I checked the records in NSE. The company was first listed somewhere in 2000. The first announcement or corporate action is from 2015. (Nothing before that).

There is also no stock with the same ISIN prior to 2015. So this is not one of those stocks.

So its definitely been actively traded since 2015. I dont know the reason for the price difference, if there was a bonus or split or other corporate action its not listed on NSE.

I'm not sure about the reliability of the broker data. TradingView shows data starting from 19th April 2018.

In this case, I prefer to stick to NSE bhavcopy version. You can ofcourse source data from the broker.

There wont be any difference during backtesting, since there is no price changes like it happens when bonus or split is declared.
